src/main/java/edu/kit/compiler/transform/IRBooleanExpressions.java

```diff
@@ -1,22 +1,272 @@
+import edu.kit.compiler.data.ast_nodes.ExpressionNode.FieldAccessExpressionNode;
+import edu.kit.compiler.data.ast_nodes.ExpressionNode.IdentifierExpressionNode;
+import edu.kit.compiler.data.ast_nodes.ExpressionNode.MethodInvocationExpressionNode;
+import edu.kit.compiler.data.ast_nodes.ExpressionNode.ThisExpressionNode;
+import edu.kit.compiler.data.ast_nodes.ExpressionNode.UnaryExpressionNode;
+import edu.kit.compiler.data.ast_nodes.ExpressionNode.ValueExpressionNode;
 import firm.Construction;
 import firm.Mode;
+import firm.Relation;
 import firm.nodes.Block;
 import firm.nodes.Node;
+import lombok.AllArgsConstructor;
 import lombok.Getter;
 
 public class IRBooleanExpressions {
+    private static final DataType boolType = new DataType(DataTypeClass.Boolean);
+
     public static Node asValue(TransformContext context, ExpressionNode expr) {
-        throw new UnsupportedOperationException();
+        if (!expr.getResultType().equals(boolType)) {
+            throw new IllegalArgumentException();
+        }
+        return expr.accept(new ValueVisitor(context));
     }
 
     public static void asConditional(TransformContext context, ExpressionNode expr,
                                      Block trueBranch, Block falseBranch) {
-        throw new UnsupportedOperationException();
+        if (!expr.getResultType().equals(boolType)) {
+            throw new IllegalArgumentException();
+        }
+        expr.accept(new ConditionalVisitor(context, trueBranch, falseBranch));
     }
 
+    @AllArgsConstructor
+    private static class ValueVisitor implements AstVisitor<Node> {
+        private TransformContext context;
+
+        @Override
+        public Node visit(BinaryExpressionNode expr) {
+            return switch (expr.getOperator()) {
+                case Assignment -> {
+                    ExpressionNode left = expr.getLeftSide();
+                    ExpressionNode right = expr.getRightSide();
+                    if (preferAsValue(right)) {
+                        Node rValue = evalExpression(context, right);
+                        throw new UnsupportedOperationException();
+                    } else {
+                        yield fromConditional(expr);
+                    }
+                }
+                default -> fromConditional(expr);
+            };
+        }
+
+        @Override
+        public Node visit(UnaryExpressionNode expr) {
+            if (expr.getOperator() != UnaryOperator.LogicalNegation) {
+                throw new IllegalArgumentException();
+            }
+
+            /**
+             * We try to create some optimized code here, because
+             * optimizing conditionals/branches afterwards is not that simple.
+             */
+            ExpressionNode inner = expr.getExpression();
+            if (inner instanceof UnaryExpressionNode) {
+                return evalExpression(context, ((UnaryExpressionNode)inner).getExpression());
```

Comment:
You could use `instanceof` pattern matching here. I believe we would need to bump the JDK version to do this though.

Reply:
And I would need to setup the new JDK version first :sweat_smile:

Reply:
Ok, fair enough.